### PR TITLE
Allow custom ports in the test runner

### PR DESCRIPTION
**Pull request by hatched**

### CODE REVIEW

**Comment by jujugui** [comment]
Refer to this link for build results (access rights to CI server needed): 
http://ci-gce.jujugui.org:8080//job/juju-gui/8142/

File: .gitignore [hunk 1]
```diff
@@ -24,6 +24,7 @@ jujugui.egg-info/
 jujugui/static/build/*
 jujugui/static/gui/build/
 karma-mocha-old.conf.js
+karma.conf.js
```
Comment (frankban) [inline .gitignore]: Is this something we generate? Just to make sure, since it wasn't excluded before.

**Comment by jujugui** [comment]
Refer to this link for build results (access rights to CI server needed): 
http://ci-gce.jujugui.org:8080//job/juju-gui/8143/

File: scripts/test-js.sh [hunk 2]
```diff
@@ -11,6 +11,12 @@ else
   PORT=8888
 fi
 
+# Modify the karma.conf.js to have the port for the proxy. This allows us to
+# use different ports in CI.
+# This needs to be placed before pserve starts or else the server will not
+# start properly.
+cp karma.conf.js.tmpl karma.conf.js
+sed -i -e 's/{TEST_PORT}/'$PORT'/' karma.conf.js
```
Comment (frankban) [inline scripts/test-js.sh]: Does this work? Why do we need quotes in '$PORT'?

File: .gitignore (hunk 1 above)
Comment (hatched) [reply .gitignore]: before we weren't generating it, it was just there. But now we only want to keep the template around.

File: scripts/test-js.sh (hunk 2 above)
Comment (hatched) [reply scripts/test-js.sh]: yes :)

**Pull request merged**